Expand ComputeFiniteStrain docs (#12384)
idaholab/moose: modules/tensor_mechanics/doc/content/source/materials/ComputeFiniteStrain.md

## Description

This class is used to compute the strain increment, total strain, and incremental rotation for finite strain problems. The finite strain approach used is the incremental
corotational form [citep:rashid1993incremental]. This approach computes logarithmic strains and strain increments.

### Incremental Configurations

In this form, the generic time increment under consideration is such that
\begin{equation}
  \label{eqn:time_notation}
  t \in [t_n, t_{n+1}]
\end{equation}
The configurations of the material element under consideration at $t = t_n$ and
$t = t_{n+1}$ are denoted by $\kappa_n$, and $\kappa_{n + 1}$, respectively for
the previous and the current incremental configurations.

### Deformation Gradient Definition

> The deformation gradient represent the change in a material element from the

Reviewer: ```suggestion
The deformation gradient represents the change in a material element from the
```

can be given in terms of $\boldsymbol{\hat{U}}$ by subsituting
[eqn:polar_decomposition_deform_grad] into the definition for $\boldsymbol{\hat{C}}$
from [cite:malvern1969introduction]:
\begin{equation}
  \label{eqn:right_green_cauchy_deformation_tensor}
  \boldsymbol{\hat{C}} = \boldsymbol{\hat{F}}^T \cdot \boldsymbol{\hat{F}} = \boldsymbol{\hat{U}}^T \cdot \boldsymbol{\hat{R}}^T \cdot \boldsymbol{\hat{R}} \cdot \boldsymbol{\hat{U}} = \boldsymbol{\hat{U}}^2
\end{equation}
where the orthogonal nature of $\boldsymbol{\hat{R}}$ enables the simplification
given above. Thus $\boldsymbol{\hat{U}}$ can be computed from $\boldsymbol{\hat{C}}$ as
\begin{equation}
  \label{eqn:definition_stretch_tensor}
  \boldsymbol{\hat{U}} = \boldsymbol{\hat{C}}^\frac{1}{2}
\end{equation}
which can be evaluated by performing a spectral decomposition of $\boldsymbol{\hat{C}}$.
Once $\boldsymbol{\hat{U}}$ has been computed, the multiplicative decomposition
of the deformation graidient is used to find the incremental rotation tensor
$\boldsymbol{\hat{R}}$ and the stretching rate $\boldsymbol{D}$.
Following [cite:rashid1993incremental], the stretching rate tensor can be expressed in terms

> of 'incremental' right Cauchy-Green deformation tensor

Reviewer: ```suggestion
of the 'incremental' right Cauchy-Green deformation tensor
```

## Description

> The finite strain mechanics approach used in the MOOSE tensor_mechanics module is the incremental

Reviewer: ```suggestion
This class is used to compute the strain increment, total strain, and incremental rotation for finite strain problems. The finite strain approach used is the incremental
```

> corotational form [citep:rashid1993incremental].

Reviewer: ```suggestion
corotational form [citep:rashid1993incremental]. This approach computes logarithmic strains and strain increments.
```

### Incremental Configurations

In this form, the generic time increment under consideration is such that
\begin{equation}
  \label{eqn:time_notation}
  t \in [t_n, t_{n+1}]
\end{equation}
The configurations of the material element under consideration at $t = t_n$ and $t = t_{n+1}$ are denoted
by $\kappa_n$, and $\kappa_{n + 1}$, respectively for the previous and the current
incremental configurations.

### Deformation Gradient

> The incremental motion over the time increment is assumed to be the inverse of

Reviewer: It seems like we should define \hat{F} first before talking about its inverse.

the deformation gradient $\hat{\boldsymbol{F}}$, that is, the deformation in
$\kappa_{n + 1}$ with respect to $\kappa_n$. This inverse deformation gradient
may be written as
\begin{equation}
  \hat{\boldsymbol{F}}^{-1} = \left( \frac{\partial{\boldsymbol{x}_{n+1}}}{\partial{\boldsymbol{x}_n}} \right)^{-1} = 1 - \frac{\partial \hat{\boldsymbol{u}}}{\partial \boldsymbol{x}},
\end{equation}
where $\hat{\boldsymbol{u}}(\boldsymbol{x})$ is the incremental displacement field for the time step, and
$\boldsymbol{x}_{n+1}$ is the position vector of materials points in $\kappa_{n+1}$.

!alert! note title=Incremental vs Total Deformation Gradient
Note that $\hat{\boldsymbol{F}}$ is NOT the deformation gradient, but rather the incremental deformation gradient
of $\kappa_{n+1}$ with respect to $\kappa_n$. Thus $\hat{\boldsymbol{F}} = \boldsymbol{F}_{n+1}
\boldsymbol{F}_n^{-1}$, where $\boldsymbol{F}_n$ is the total deformation gradient at time $t_n$.
!alert-end!

> As shown in [cite:zhang2018modified], the incremental deformation gradient can

Reviewer: ```suggestion
Following the explanation of this procedure given by [cite:zhang2018modified], the incremental deformation gradient can
```

Reviewer: I didn't want to make it sound like we invented it!

> be multiplicatly decomposed into an incremental rotation tensor, $\boldsymbol{\hat{R}}$,

Reviewer: ```suggestion
be multiplicatively decomposed into an incremental rotation tensor, $\boldsymbol{\hat{R}}$,
```

> and the incremental right stretch tensor, $\boldsymbol{\hat{U}}

Reviewer: ```suggestion
and the incremental right stretch tensor, $\boldsymbol{\hat{U}}$
```

\end{equation}
where the orthogonal nature of $\boldsymbol{\hat{R}}$ enables the simplification
given above. Thus $\boldsymbol{\hat{U}}$ can be computed from $\boldsymbol{\hat{C}}$ as
\begin{equation}
  \label{eqn:definition_stretch_tensor}
  \boldsymbol{\hat{U}} = \boldsymbol{\hat{C}}^\frac{1}{2}
\end{equation}
which can be evaluated by performing a spectral decomposition of $\boldsymbol{\hat{C}}$.
Once $\boldsymbol{\hat{U}}$ has been computed, the multiplicative decomposition
of the deformation graidient is used to find the incremental rotation tensor
$\boldsymbol{\hat{R}}$ and the stretching rate $\boldsymbol{D}$.
Following [cite:rashid1993incremental], the stretching rate tensor can be expressed in terms
of 'incremental' right Cauchy-Green deformation tensor
\begin{equation}
  \label{eqn:stretching_tensor_definition}
  \boldsymbol{D} = \frac{1}{\Delta t}\log({\hat{\boldsymbol{C}}^{1/2}})
\end{equation}

This incremental streteching rate tensor can then be used as the work conjugate

> for a stress measure, as shown in [cite:zhang2018modified], or used to find a strain

Reviewer: ```suggestion
for a stress measure, or used to compute another strain
```

> measure. Therefore `ComputeFiniteStrain` includes two different methods to solve

Reviewer: ```suggestion
measure. The most computationally expensive part of this procedure is the spectral decomposition of $\boldsymbol{\hat{C}}$ to find $\boldsymbol{\hat{U}}$. This can be computed exactly using an [eigensolution](#eigensolution), but an approximation of this can be computed with much lower computational expense using a [Taylor expansion](#taylorexpansion) procedure. This class provides options to perform this calculation either way, but the Taylor expansion is the default.
```

Reviewer: Delete the two lines below if you like that text.

for the stretching rate tensor: the [Taylor Expansion](#taylorexpansion) and the
[Eigenstrain Decomposition](#eigensolution). The Taylor Expansion is the default.

> ### Assumptions for the Incremental Deformation Gradient Formulation

Reviewer: This section feels a little out of place. Maybe we should just combine this stuff with the previous section.

Author: okay. I remember we had talked about moving it, but I couldn't remember where to move the section

Author: Note to self: add a sentence about consistency in the kernel with `use_displaced_mesh` below the finite strain example block

\end{aligned}
\end{equation}

These assumptions are used the Taylor Expansion to solve for the stretching rate
and rotation tensors.

## Taylor Expansion id=taylorexpansion

> The stretching rate tensor $\boldsymbol{D}$ and rotation matrix $\boldsymbol{R}$

Reviewer: ```suggestion
The stretching rate tensor $\boldsymbol{D}$ and incremental rotation matrix $\hat{\boldsymbol{R}}$
```
